gatorgrade/input/in_file_path.py

```diff
@@ -7,3 +7,28 @@ def parse_yaml_file(file_path):
     with open(file_path, encoding="utf8") as file:
         data = yaml.load_all(file, Loader=yaml.FullLoader)
         return list(data)
+
+
+def reformat_yaml_data(data):
+    """Reformat the raw data from a YAML file into a list of tuples."""
+    reformatted_data = []
+    data.pop(0)  # Removes the setup commands
+    add_checks_to_list(None, data[0], reformatted_data)
+    return reformatted_data
+
+
+def add_checks_to_list(path, data_list, reformatted_data):
+    """Recursively loop through the data and add any checks that are found to the reformatted list."""
+    current_path = path
+    for dict in data_list:
+        for item in dict:
+            if isinstance(dict[item], list):
+                if not path:
+                    path = item
+                else:
+                    path = f"{path}/{item}"
+                add_checks_to_list(path, dict[item], reformatted_data)
+                path = current_path
+            else:
+                reformatted_data.append({"file_context": path, "check": dict})
+                break
```

Comment:
This code all looks pretty good! The only think I'm noticing is an overall lack of source code comments. This function specifically has none. Source code comments are a great method of documentation and can make the code easier to understand especially for those who might find themselves revisiting this in the future. If you could take some time to write some useful comments I think it would greatly improve the sustainability of this work. Otherwise, good work!

Reply:
I just added some comments to this function that will hopefully make it better documented. Let me know if you think anything else could be added to improve the documentation in the code.

Reply:
The added comments look great, thank you for taking the time to work them in.